arangod/Agency/State.cpp

```diff
@@ -854,12 +855,35 @@ bool State::loadPersisted() {
   loadOrPersistConfiguration();
 
   if (checkCollection("log") && checkCollection("compact")) {
-    bool lc = loadCompacted();
-    bool lr = loadRemaining();
-    return (lc && lr);
+    index_t lastCompactionIndex = loadCompacted();
+    if (loadRemaining(lastCompactionIndex)) {
+      return true;
+    } else {
+      LOG_TOPIC("1a476", INFO, Logger::AGENCY)
+        << "Non matching compaction and log indexes. Dropping both collections";
+
+      std::function<void(std::string const&)> drop = [&](std::string const& colName) {
+        try {
+          auto col = _vocbase->lookupCollection(colName)->id();
+          auto res = _vocbase->dropCollection(col, false, -1.0).errorNumber();
+          if (res != TRI_ERROR_NO_ERROR) {
+            LOG_TOPIC("ba841", ERR, Logger::AGENCY)
+              << "unable to drop collection log: " << TRI_errno_string(res);
+          }
+        } catch (std::exception const& e) {
+          LOG_TOPIC("69fc4", FATAL, Logger::REPLICATION)
+            << "unable to drop collections log/compact" << e.what();
+        }
+      };
+      _log.clear();
+      _cur = 0;
+      drop("log");
+      drop("compact");
```

Comment:
If we crash here, then `log` might be gone but then `compact` is still there. I think we have a problem that in this case the next start does not work. See other comment.

Reply:
Cannot comment there, so let's do it here: In `State::loadPersisted` wird es knallen, wenn nur eine der beiden Collections da ist. Denn dann wird so getan, als ob keine da ist. Und dann knallt es, wenn die, die doch da ist, angelegt werden soll.

Reply:
No that's fine too. Next time around we run into `createCollections` should take care of that. We should fix that code.